facebook-commerce.php

```diff
@@ -1122,13 +1122,21 @@ function on_variable_product_publish( $wp_id, $woo_product = null ) {
 			$this->create_product_group( $woo_product, $retailer_id, true );
 		}
 
-		$child_products = $woo_product->get_children();
+		$variation_ids = [];
 
-		// scheduled update for each variation
-		foreach ( $child_products as $item_id ) {
+		// scheduled update for each variation that should be synced
+		foreach ( $woo_product->get_children() as $variation_id ) {
 
-			facebook_for_woocommerce()->get_products_sync_handler()->create_or_update_products( [ $item_id ] );
+			if ( ! $variation = wc_get_product( $variation_id ) ) {
+				continue;
+			}
```

Comment:
```suggestion
			$variation = wc_get_product( $variation_id )
```

later you are checking if `$variation` is an instance of `\WC_Product` so the condition check with the variable assignment is redundant